src/libs/actions/IOU.js

```diff
@@ -1121,44 +1126,46 @@ function sendMoneyViaPaypal(report, amount, currency, comment, managerEmail, rec
 }
 
 /**
+ * @param {String} paymentType
  * @param {Object} chatReport
  * @param {Object} iouReport
  * @param {Object} recipient
+ * @param {String} reimbursementBankAccountState
  */
-function payMoneyRequestElsewhere(chatReport, iouReport, recipient) {
-    const {params, optimisticData, successData, failureData} = getPayMoneyRequestParams(chatReport, iouReport, recipient, CONST.IOU.PAYMENT_TYPE.ELSEWHERE);
-
-    API.write('PayMoneyRequestElsewhere', params, {optimisticData, successData, failureData});
-
-    Navigation.navigate(ROUTES.getReportRoute(chatReport.reportID));
-}
-
-/**
- * @param {Object} chatReport
- * @param {Object} iouReport
- * @param {Object} recipient
- */
-function payMoneyRequestWithWallet(chatReport, iouReport, recipient) {
-    const {params, optimisticData, successData, failureData} = getPayMoneyRequestParams(chatReport, iouReport, recipient, CONST.IOU.PAYMENT_TYPE.EXPENSIFY);
-
-    API.write('PayMoneyRequestWithWallet', params, {optimisticData, successData, failureData});
-
-    Navigation.navigate(ROUTES.getReportRoute(chatReport.reportID));
-}
-
-/**
- * @param {Object} chatReport
- * @param {Object} iouReport
- * @param {Object} recipient
- */
-function payMoneyRequestViaPaypal(chatReport, iouReport, recipient) {
-    const {params, optimisticData, successData, failureData} = getPayMoneyRequestParams(chatReport, iouReport, recipient, CONST.IOU.PAYMENT_TYPE.PAYPAL_ME);
-
-    API.write('PayMoneyRequestViaPaypal', params, {optimisticData, successData, failureData});
+function payMoneyRequest(paymentType, chatReport, iouReport, reimbursementBankAccountState) {
+    if (
+        chatReport.chatType === CONST.REPORT.CHAT_TYPE.POLICY_EXPENSE_CHAT &&
+        reimbursementBankAccountState !== CONST.BANK_ACCOUNT.STATE.OPEN &&
+        paymentType === CONST.IOU.PAYMENT_TYPE.EXPENSIFY
+    ) {
+        Navigation.navigate(ROUTES.BANK_ACCOUNT_NEW);
+        return;
+    }
 
+    const recipient = {
+        login: iouReport.ownerEmail,
+        payPalMeAddress: iouReport.submitterPayPalMeAddress,
+    };
+    const {params, optimisticData, successData, failureData} = getPayMoneyRequestParams(chatReport, iouReport, recipient, paymentType);
+    let command;
+    switch (paymentType) {
+        case CONST.IOU.PAYMENT_TYPE.PAYPAL_ME:
+            command = 'PayMoneyRequestViaPaypal';
+            break;
+        case CONST.IOU.PAYMENT_TYPE.EXPENSIFY:
+            command = 'PayMoneyRequestWithWallet';
+            break;
+        case CONST.IOU.PAYMENT_TYPE.ELSEWHERE:
+            command = 'PayMoneyRequestElsewhere';
+            break;
+        default:
+            throw new Error('Unsupported payment method');
```

Comment:
@mountiny are we still using these commands on the backend?

Reply:
Its used in PHP but not in Auth, we can refactor this later, this should be work fine